File: ejercicios hechos/Python/estructuras alternativas/act10.py
#Programa: act10.py
#Propósito: clasificar unas circunferencias
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 17/10/2019
#
#Variables a usar:
# x1,y1,x2,y2 y r1,r2 son los puntos de la circunferencia
# d es la distancia entre los centros de las circunferencias
#
#Algoritmo:
# LEER x1,y1 x2,y2 y r1,r2
# Calculamos d <-- ((x2 - x1) ** 2 + (y2 - y1) ** 2) ** 1/2
# si d es mayor que la suma de r1 y r2 son exteriores
# si d es menor que la suma y mayor que la diferencia son secantes
# si d es mayor que 0 pero menor que la diferencia son interiores
# si d es igual a la suma de los radios son Tangentes exteriores
# si d es igual a la diferenfcia entre los radios son Tangentes interiores
# si d es igual a 0 son concentricas

print("Clasificar dos circunferencias relativas a su posición")
print("-------------------------------------------------------\n")

#Leemos los datos
x1 = int(input("Introduce el punto x1: "))
y1 = int(input("Introduce el punto y1: "))
r1 = int(input("Introduce el punto r1: "))
x2 = int(input("Introduce el punto x2: "))
y2 = int(input("Introduce el punto y2: "))
r2 = int(input("Introduce el punto r2: "))

# Calculamos d
d = ((x2 - x1) ** 2 + (y2 - y1) ** 2) ** 1/2

# realizamos las comparativas
if d > r1 + r2:
    print("Las circunferencias son exteriores")
    
elif d < r1 + r2 and d > abs(r1 - r2):
    print("Las circunferencias son secantes")
    
elif d < 0 and d < abs(r1 - r2):
    print("Las circunferencias son interiores")

elif d == r1 + r2:
    print("Las circunferencias son Tangentes exteriores")

elif d == abs(r1 - r2):
    print("Las circunferencias son Tangentes interiores")
    
elif d == 0:
    print("Las circunferencias son concentricas")

File: ejercicios terminados/Java/estructuras alternativas/act10.java
/*Programa: act10.java
 * 
 *Propósito: Algoritmo que pida los puntos centrales x1,y1,x2,y2 y los radios r1,r2 de dos 
circunferencias y las clasifique en uno de estos estados:

exteriores
tangentes exteriores
secantes
tangentes interiores
interiores
concéntricas
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 09/11/2019
 *
 *Análisis:
 * Nos dan un centro y un radio de dos circunferencias
 * debemos comprobar su organización en el plano
 *
 *Diseño:
 * 1. Leer x1,y1,r1 x2,y2,r2
 * 2. Caculamos la distancia entre los centros
 * 4. Si distancia>suma de los radios mostrar "Circunferencias exteriores"
 * 5. Si distancia = suma de los radios mostrar "Tangentes exteriores"
 * 6. Si distancia < suma de los radios Y distancia> que el valor absoluto de la resta de los radios mostrar "Circunferencias secante"
 * 7. Si distancia = que el valor absoluto de la resta de los radios mostrar "Circunferencias tangentes interiores"
 * 8. Si distancia >0 y distancia < que el valor absoluto de la resta de los radios mostrar "Circunferencias interiores"
 * 9. Si distancia = 0  mostrar "Circunferencias concéntricas"
*/
import java.util.Scanner;
 
public class act10 {
  public static void main(String args[]) {
    Scanner s = new Scanner(System.in);
    System.out.println("?");
    System.out.println("--------------------------------------------------");
    double distancia;
    double r1;
    double r2;
    double x1;
    double x2;
    double y1;
    double y2;

    System.out.print("Dime coordenada x primera circunferencia: ");
    x1 = s.nextDouble();
    System.out.print("Dime coordenada y primera circunferencia: ");
    y1 = s.nextDouble();
    System.out.print("Dime radio primera circunferencia: ");
    r1 = s.nextDouble();
    System.out.print("Dime coordenada x segunda circunferencia: ");
    x2 = s.nextDouble();
    System.out.print("Dime coordenada y segunda circunferencia: ");
    y2 = s.nextDouble();
    System.out.print("Dime radio segunda circunferencia: ");
    r2 = s.nextDouble();

    distancia = Math.sqrt(Math.pow((x2-x1),2)+Math.pow((y2-y1),2));
    
    if (distancia>(r1+r2)) {
      System.out.println("Circunferencias exteriores");
    }
    if (distancia==(r1+r2)) {
      System.out.println("Circunferencias tangentes exteriores");
    }
    if (distancia<(r1+r2) && distancia>Math.abs(r1-r2)) {
      System.out.println("Circunferencias secantes");
    }
    if (distancia==Math.abs(r1-r2)) {
      System.out.println("Circunferencias tangentes interiores");
    }
    if (distancia>0 && distancia<Math.abs(r1-r2)) {
      System.out.println("Circunferencias interiores");
    }
    if (distancia==0) {
      System.out.println("Circunferencias concétricas");
    }
  }


}
